### ISSUE TITLE
Add analyzer to Razor.Diagnostic.Analzyers that detects scenarios where a pooled object or pooled array escapes its scope

### ISSUE THREAD

**Issue by DustinCampbell** [dotnet/roslyn#85560]
Common Razor object pooling patterns should be detectable with an analyzer. For example, for any `PooledObject<T>` declared within a using statement, it should be possible to detect scenarios where the object is allowed to escape beyond the using's scope. For example, in the following code, an analyzer could detect the fact `list` escapes its scope and report an error.

```C#
List<int> M()
{
    using var _ = ListPool<int>.GetPooledObject(out var list);
    
    // use list...

    return list;
}
```
